cmd/internal/cli/actions.go

```diff
@@ -290,11 +294,35 @@ func handleShub(imgCache *cache.Handle, u string) (string, error) {
 }
 
 func handleNet(imgCache *cache.Handle, u string) (string, error) {
-	refParts := strings.Split(u, "/")
-	imageName := refParts[len(refParts)-1]
-	imagePath := imgCache.NetImage("hash", imageName)
+	// We will cache using a sha256 over the URL and the date of the file that
+	// is to be fetched, as returned by an HTTP HEAD call and the Last-Modified
+	// header. If no date is available, use the current date-time, which will
+	// effectively result in no caching.
+	imageDate := time.Now().String()
 
-	exists, err := imgCache.NetImageExists("hash", imageName)
+	req, err := http.NewRequest("HEAD", u, nil)
```

Comment:
Maybe trim `http://`, `https://` from the hash as some server allow access via http and/or redirect http to https

Reply:
That's true - but equally a crazy setup could serve different things on http and http(s). That would be really stupid and unlikely, but I'd rather err on the side of re-downloading an image unless you feel strongly about this?

Reply:
Not really it was just a suggestion